pkg/proxy/ipvs/proxier.go

```diff
@@ -1328,17 +1328,22 @@ func (proxier *Proxier) syncEndpoint(svcPortName proxy.ServicePortName, onlyNode
 }
 
 func (proxier *Proxier) cleanLegacyService(atciveServices map[string]bool, currentServices map[string]*utilipvs.VirtualServer) {
+	unbindIPAddr := map[string]bool{}
 	for cS := range currentServices {
 		if !atciveServices[cS] {
 			svc := currentServices[cS]
 			err := proxier.ipvs.DeleteVirtualServer(svc)
 			if err != nil {
 				glog.Errorf("Failed to delete service, error: %v", err)
 			}
-			err = proxier.ipvs.UnbindVirtualServerAddress(svc, DefaultDummyDevice)
-			if err != nil {
-				glog.Errorf("Failed to unbind service from dummy interface, error: %v", err)
-			}
+			unbindIPAddr[svc.Address.String()] = true
+		}
+	}
+
+	for addr := range unbindIPAddr {
+		err := proxier.ipvs.UnbindVirtualServerAddress(addr, DefaultDummyDevice)
+		if err != nil {
+			glog.Errorf("Failed to unbind service from dummy interface, error: %v", err)
```

Comment:
output IP address will help debug

Reply:
More debug info will be added in next commit.

Reply:
ACK